sklearn/ensemble/tests/test_iforest.py

```diff
@@ -58,9 +58,9 @@ def test_iforest():
             IsolationForest(random_state=rng, **params).fit(X_train).predict(X_test)
 
 
-def test_iforest_sparse():
+def test_iforest_sparse(global_random_seed):
```

Comment:
In this test, IsolationForest is still instanciated with random_state=1. I think it would make sense to set `random_state=global_random_seed`